clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -750,6 +750,8 @@ public void handleCompletedMetadataResponse(RequestHeader requestHeader, long no
             Cluster cluster = response.cluster();
             // check if any topics metadata failed to get updated
             Map<String, Errors> errors = response.errors();
+            metadata.lastErrors(errors);
```

Comment:
Was this required to get your authorization integration test to work? If so, why?